src/main/java/frc/robot/subsystems/TurretSubsystem.java

```diff
@@ -141,33 +227,58 @@ private void setTurretPos(double setpoint) {
             turretPosition,
             MathUtil.clamp(
                 setpoint,
-                Math.toRadians(-Constants.TurretConstants.MAX_TURRET_ANGLE_DEGREES),
-                Math.toRadians(Constants.TurretConstants.MAX_TURRET_ANGLE_DEGREES))));
+                Math.toRadians(-Constants.TurretConstants.TURRET_MAX_ANGLE_DEGREES),
+                Math.toRadians(Constants.TurretConstants.TURRET_MAX_ANGLE_DEGREES))));
+  }
+
+  private void setPitchPos(double setpoint) {
+    pitchGoalPositionEntry.log(setpoint);
+    pitchMotor.set(
+        pitchPIDController.calculate(
+                pitchPosition,
+                MathUtil.clamp(
+                    setpoint,
+                    Math.toRadians(-Constants.TurretConstants.PITCH_MAX_ANGLE_DEGREES),
+                    Math.toRadians(Constants.TurretConstants.PITCH_MAX_ANGLE_DEGREES)))
+            + calculateArbitraryFeedForward(pitchPosition));
   }
 
   private void holdPosition() {
     setTurretPos(turretPosition);
+    setPitchPos(pitchPosition);
   }
 
   public Command idleTurret() {
     return run(() -> holdPosition()).withName("idleTurret");
   }
 
+  private void moveUp() {
+    setTurretPos(turretPosition);
+    setPitchPos(pitchPosition + 0.5);
+  }
+
+  public Command moveUpwards() {
+    return run(() -> moveUp()).withName("moveShooterUp");
+  }
+
   private void zeroTurretEncoder() {
-    m_encoder.setPosition(0.0);
+    m_turretEncoder.setPosition(0.0);
   }
 
   private void updateTurretPosition() {
     turretPosition =
         Math.toRadians(
-            ((m_encoder.getPosition().getValueAsDouble()) * 360)
-                * Constants.TurretConstants.CONVERSION_FACTOR);
+            ((m_turretEncoder.getPosition().getValueAsDouble()) * 360)
+                * Constants.TurretConstants.TURRET_CONVERSION_FACTOR);
     SmartDashboard.putNumber("Turret Position", turretPosition);
-    SmartDashboard.putBoolean("Out of Bounds", Math.abs(turretPosition) > 3.14);
-    SmartDashboard.putBoolean(
-        "Soft limit enabled forward", turretMotor.isSoftLimitEnabled(SoftLimitDirection.kForward));
-    SmartDashboard.putBoolean(
-        "Soft limit enabled reverse", turretMotor.isSoftLimitEnabled(SoftLimitDirection.kReverse));
+  }
+
+  private void updatepitchPosition() {
+    pitchPosition =
+        Math.toRadians(
+            ((m_pitchEncoder.getPosition().getValueAsDouble()) * 360)
+                * Constants.TurretConstants.PITCH_CONVERSION_FACTOR);
+    SmartDashboard.putNumber("Pitch Position", pitchPosition);
```

Comment:
Use DebugEntry for this?  So that we log it even when we are in competition?

CHange name of method to be updatePitchPosition

I also don't see this called?  Should this be in periodic?

Reply:
It's now in periodic and a DebugEntry.